File: kbmtl_semisupervised_regression_variational_test.R
# Mehmet Gonen (mehmet.gonen@gmail.com)

kbmtl_semisupervised_regression_variational_test <- function(K, state) {
  N <- dim(K)[2]
  T <- dim(state$W$mu)[2]

  H <- list(mu = crossprod(state$A$mu, K))

  Y <- list(mu = crossprod(H$mu, state$W$mu), sigma = matrix(0, N, T))
  for (t in 1:T) {
    Y$sigma[,t] <- 1 / (state$epsilon$alpha[t] * state$epsilon$beta[t]) + diag(crossprod(H$mu, state$W$sigma[,,t]) %*% H$mu)
  }

  prediction <- list(H = H, Y = Y)
}

File: kbmtl_semisupervised_regression_variational_test.m
% Mehmet Gonen (mehmet.gonen@gmail.com)

function prediction = kbmtl_semisupervised_regression_variational_test(K, state)
    N = size(K, 2);
    T = size(state.W.mu, 2);

    prediction.H.mu = state.A.mu' * K;

    prediction.Y.mu = prediction.H.mu' * state.W.mu;
    prediction.Y.sigma = zeros(N, T);
    for t = 1:T
        prediction.Y.sigma(:, t) = 1 / (state.epsilon.alpha(t) * state.epsilon.beta(t)) + diag(prediction.H.mu' * state.W.sigma(:, :, t) * prediction.H.mu);
    end
end
